### ISSUE TITLE
bindName breaks metadata for stache helpers passed as view model methods

### ISSUE THREAD

**Issue by rjgotten** [canjs/can-stache-key#50]
It's possible to define custom stache helpers with metadata properties such as `ignoreArgLookup`; `requiresOptionsArgument` etc. and then to access and use these locally only, without registering them as global helpers, simply by putting them onto the viewmodel as methods.

However, attempting to access these as `this.<helpername>` will lead to a `BoundFunctionObject` having been created by `bindName` via the `observeReader` with `name:"function"` where `options.proxyMethods` is undefined and thus calls into `bindName`.

`bindName` uses `Function.prototype.bind` to correctly bind the function to its source as a method, to ensure `this` within the function is correct. However, it fails to copy all the special metadata qualities for helpers onto the resulting `BoundFunctionObject` and this leads to the helper-specific functionality _being broken_.

This is especially painful when using inline partials (e.g. `{{<inlined}}content{{/inlined>`), as those don't have access to the ambient top-level scope. They require everything be passed in. But you can't pass in _just_ the helper, because then it gets aliased into a `BoundFunctionObject` and breaks. Instead, to get access to the helpers from the view model inside the partial, you basically have to pass in the _entire_ viewModel and then directly call the helper inside the partial without any form of aliasing or again, it breaks.

E.g.
```
{{#my-helper}}this is fine{{/my-helper}}
{{broken(my-helper = my-helper)}}
{{workaround(root = this)}}

{{<broken}}
  {{#this.my-helper}}
     may be broken if depending on special helper metadata
  {{/my-helper}}
{{/broken}}

{{<workaround}}
  {{#this.root.my-helper}}
    functioning but verbose workaround
  {{/my-helper}}
{{/workaround}}
```

Honestly - it feels like `bindName` should just copy all 'own properties' that are on the function and that will fix it.

**Comment by justinbmeyer** [comment]
Yeah, that makes sense to me. If you can submit a PR, I'll get it released. Otherwise, I'll find some time in the coming weeks (I'm about to start a new project next week and expect to be busy).  Maybe I can ask @bmomberger-bitovi to help too :-)

**Comment by rjgotten** [comment]
I was planning on cooking up a local patch based on the `patch-package` NPM package as at least a stop-gap solution, but I'll have to see if I have time to turn it into a full blown PR. (Stuck in a busy period myself as well.)
